Svc/FileUplink/File.cpp

```diff
@@ -40,13 +40,16 @@ namespace Svc {
 
     Os::File::Status status;
     status = this->osFile.seek(byteOffset);
-    if (status != Os::File::OP_OK)
-      return status;
+    if (status != Os::File::OP_OK) {
+        return status;
+    }
 
     NATIVE_INT_TYPE intLength = length;
-    status = this->osFile.write(data, intLength);
-    if (status != Os::File::OP_OK)
-      return status;
+    //Note: not waiting for the file write to finish
+    status = this->osFile.write(data, intLength, false);
```

Comment:
Could changing write behavior to not block cause any side affects?

Reply:
I don't believe so, as this wasn't a true non-block operation before.  This just doesn't force a "sync" call through the OS before moving on. `sync` calls are fairly heavy weight and rarely needed these days.  They do however fight with the OSes ability to buffer files.